### PR TITLE
Refactor onlyoffice_docker

### PR DESCRIPTION
**Pull request by szaimen**
As the title says.
It allows to reuse a lot of codes in all office scripts
And it will allow to install different office soultions at the same time.

### CODE REVIEW

File: apps/onlyoffice_docker.sh [hunk 1]
```diff
@@ -7,7 +7,7 @@ true
 SCRIPT_NAME="OnlyOffice (Docker)"
 SCRIPT_EXPLAINER="This script will install the OnlyOffice Document Server bundled with Docker"
 # shellcheck source=lib.sh
-source /var/scripts/fetch_lib.sh || source <(curl -sL https://raw.githubusercontent.com/nextcloud/vm/master/lib.sh)
+source <(curl -sL https://raw.githubusercontent.com/nextcloud/vm/szaimen-patch-22/lib.sh)
```
Comment (szaimen) [inline apps/onlyoffice_docker.sh:10]: TODO: this needs to get reverted before merging

File: lib.sh [hunk 2]
```diff
@@ -840,6 +889,27 @@ Please follow this guide to open ports in your router or firewall:\nhttps://www.
...
 
+domain_flow() {
+    SUBDOMAIN=$(input_box_flow "$1 subdomain e.g: office.yourdomain.com
+NOTE: This domain must be different than your Nextcloud domain. \
+They can however be hosted on the same server, but would require seperate DNS entries.")
+
+    # shellcheck disable=2034,2059
+    true
+    # shellcheck source=lib.sh
+    source <(curl -sL https://raw.githubusercontent.com/nextcloud/vm/szaimen-patch-22/lib.sh)
```
Comment (szaimen) [inline lib.sh:900]: TODO: this needs to get reverted before merging

**Comment by szaimen** [comment]
@enoch85 I've successfully tested this.
Please have a look yourself.

We still need to fix the correctt links to master library before merging, though.

**Comment by enoch85** [comment]
Just tested.

1. Installed onlyoffice
2. Installed collabora --> onlyoffice removed as usual
3. installer onlyoffice --> new popup appeard (shouldn't that appear in collabora as well?)

**Comment by enoch85** [comment]
The last commit should be a PR, this is broken now. But I'm too tired to continue.

I hope you get the idea and can continue to work on it.

Thanks!

File: lib.sh [hunk 3]
```diff
@@ -1055,6 +1179,25 @@ else
...
+# OnlyOffice:
+# 1=richdocuemnts
+# 2=documentserver
+# 3=onlyoffice
+    if is_app_enabled "$1"
+    then
+        msg_box "$2 integration is enabled and will get disabled now.
+You can enable it later again over the appstore."
+        nextcloud_occ app:remove "$1" ## need a better fucntion here
+        docker_prune_this '"$3"/"$2"'
```
[the remark this answers is not in the document]
Comment (enoch85) [reply lib.sh:1197]: @szaimen wooop :D

**Pull request closed without merging**